### PR TITLE
ZIR-215: abstract bibc file format reader & writer for use on memory buffers

### PR DESCRIPTION
**Pull request by mars-risc0**
instead of generating a tmp file, you can now read from/write to a memory buffer

### CODE REVIEW

File: zirgen/Dialect/BigInt/Bytecode/test/test.cpp [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+    // Drop the old bytecode structure and create a fresh one
+    prog.reset(new BigInt::Bytecode::Program);
+    // Read the contents of the buffer back in
+    BigInt::Bytecode::read(*prog, buf.get(), bytes);
+    // Decode the bytecode back into MLIR operations
+    return BigInt::Bytecode::decode(module, *prog);
+  }
+
+  void lower() {
+    // Lower the inverse and reduce ops to simpler, executable ops
```
Comment (tzerrell) [inline zirgen/Dialect/BigInt/Bytecode/test/test.cpp:80]: Did you forget to add `createLowerInvPass` in this function? Or is there a reason it's not here?

**Review by tzerrell — COMMENTED** [review]
Looks good! Marked one possible oversight inline.

File: zirgen/Dialect/BigInt/Bytecode/test/test.cpp (hunk 1 above)
Comment (mars-risc0) [reply zirgen/Dialect/BigInt/Bytecode/test/test.cpp:80]: The RSA test cases work fine without it. This test apparatus would certainly be more generally useful if it were included, though, so I'll add that call.

**Review by tzerrell — APPROVED** [review]
Thanks!

**Pull request merged**